samples/interactive-tutorials/src/main/java/search/tutorials/search_with_boost_spec.md

```diff
@@ -0,0 +1,219 @@
+    ```bash
+    cd java-retail/samples/interactive-tutorials
+    ```
+
+## Boost by one criterion: condition and filtering expression
+
+The boosting specification looks like this:
+
+```
+  condition_boost_specs {
+         condition: string
+         boost: double [-1;1]
+  }
+```
+
+1. To set the **```condition```**, you should use a filtering expression like the following:
+
+   ```"(colorFamilies: ANY("Blue"))"``` or ```"(rating: IN(4.0, 5.0))"```
+
+<!-- TODO(ianan): change Filtering Tutorial link -->
+   You can learn how to use filters in the [Filtering Tutorial](retail_api_v2_filtering_python.md)
+   or read about it in the [Retail API documentation](https://cloud.google.com/retail/docs/filter-and-order#filter)
+
+   The field **```boost```** defines the strength of the condition boost, which should be in the range of -1 to 1. Negative boost means demotion.
+
+1. Now open <walkthrough-editor-select-regex filePath="cloudshell_open/interactive-tutorials/search/SearchWithBoostSpec.java" regex="boost.*0">SearchWithBoostSpec.java</walkthrough-editor-select-regex>.
+
+
+    In the initial request, the boost strength is set to zero: ```boost = 0.0```,
+    so the boosting will not affect the order of the   products in the response.
+
+
+## Boost by one criterion: Boost results
+
+1. Run the sample in a terminal with the following command:
+    ```bash
+    mvn compile exec:java -Dexec.mainClass="search.SearchWithBoostSpec"
+    ```
+1. Check the response to see the original order of products depending on their relevance to the query phrase.
+
+1. Next, change the value of the field **boost** and run the code sample again:
+   ```boost = 1.0```
+
+1. Now you can check the results. The products corresponding to the boost condition became reranked. Now blue products are on the top of the list.
+
+1. If you set ```boost = -1.0```, blue products will appear at the bottom of the search result.
+
+## Some notes about boosting
+
+Setting the boost strength to 1.0 gives the item a strong promotion. However, it does not necessarily mean that the boosted item will be the top result at all times, nor that other items will be excluded.
+Results could still be shown even when none of them matches the condition.
+
+Also, results that are significantly more relevant to the search query can still be ranked higher than your heavily favored but irrelevant items.
+
+You can combine up to 10 boost specifications in one search request. In this way, you can apply really sophisticated boosting rules to your search request.
+
+## Try different boosting conditions
+
+Feel free to test product boosting yourself right now in the Cloud Shell environment.
+
+Replace the <walkthrough-editor-select-regex filePath="cloudshell_open/interactive-tutorials/search/SearchWithBoostSpec.java" regex="condition = '.*'">condition</walkthrough-editor-select-regex> expression with something like this:
```

Comment:
I think it's not the full path 
Example: interactive-tutorials/src/main/java/search/SearchSimpleQuery.java

Reply:
Done for all *.md files.